### DISCUSSION TITLE
How do I let a bot recognize users?

### DISCUSSION THREAD

**Discussion by MechanicalDeus** [hikari-py/hikari#1365] [Q&A] [answered]
I would like to code some games for Discord Bots.
While I worked on a Pen&Paper Bot for Character Editing I wondered
how I could let a bot listen to a specific User by ID and react with that users instance
of Characters, events ect. As for now my bot will only do something to or by a specific character
if its name is specified in the command, which allows basically anybody to control any character.
But I would like it to be that a user is recognized and his character
is automatically chosen to do the actions.

for further explanation: I use lightbulb for hikari wich is for but not of hikari.
It just seems to be an extension that adds slashcommands and subgroups of them.
Thats what I use it for.

So if a user types /use "something specific"

I want the bot to automatically fetch that Users ID and check if the User has been registered, and then to react with
that users specific Character or instance of the game.

I figured that I need to construct a User class that contains their User ID to recognize them but im quite new to
Discord Bots so Id be glad if someone knows how to do it.

For example there could be real time games like these old Browser Games with timers and set actions you need to confirm.
I would  really like to try something like that, but thats what I need User recognization for.

**Comment by davfsa** [comment, marked as answer]
Something you can do is link a user ID to its "character" and store that information in a database. You can get the user that executed the interaction from the event that is received, and I'm sure lightbulb exports that in their context variable (iirc, its `ctx.author.id`).

Hope that answers  your question. If you have any follow-up ones, you can ask them here and I will try my best to answer or  you can join our Discord server and ask there (https://discord.gg/hikari) :)
